spec.html

```diff
@@ -46491,27 +46575,465 @@ <h1>The <i>IteratorResult</i> Interface</h1>
       </emu-clause>
     </emu-clause>
 
-    <emu-clause id="sec-%iteratorprototype%-object">
-      <h1>The %IteratorPrototype% Object</h1>
-      <p>The <dfn>%IteratorPrototype%</dfn> object:</p>
+    <emu-clause id="sec-iterator-helper-objects">
+      <h1>Iterator Helper Objects</h1>
+      <p>An Iterator Helper object is an ordinary object that represents a lazy transformation of some specific source iterator object. There is not a named constructor for Iterator Helper objects. Instead, Iterator Helper objects are created by calling certain methods of Iterator instance objects.</p>
+
+      <emu-clause id="sec-%iteratorhelperprototype%-object">
+        <h1>The %IteratorHelperPrototype% Object</h1>
+        <p>The <dfn>%IteratorHelperPrototype%</dfn> object:</p>
+        <ul>
+          <li>has properties that are inherited by all Iterator Helper Objects.</li>
+          <li>is an ordinary object.</li>
+          <li>has a [[Prototype]] internal slot whose value is %Iterator.prototype%.</li>
+          <li>has the following properties:</li>
+        </ul>
+
+        <emu-clause id="sec-%iteratorhelperprototype%.next">
+          <h1>%IteratorHelperPrototype%.next ( )</h1>
+          <emu-alg>
+            1. Return ? GeneratorResume(*this* value, *undefined*, *"Iterator Helper"*).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-%iteratorhelperprototype%.return">
+          <h1>%IteratorHelperPrototype%.return ( )</h1>
+          <emu-alg>
+            1. Let _O_ be *this* value.
+            1. Perform ? RequireInternalSlot(_O_, [[UnderlyingIterator]]).
+            1. Assert: _O_ has a [[GeneratorState]] internal slot.
+            1. If _O_.[[GeneratorState]] is ~suspended-start~, then
+              1. Set _O_.[[GeneratorState]] to ~completed~.
+              1. NOTE: Once a generator enters the completed state it never leaves it and its associated execution context is never resumed. Any execution state associated with _O_ can be discarded at this point.
+              1. Perform ? IteratorClose(_O_.[[UnderlyingIterator]], ReturnCompletion(*undefined*)).
```

Comment:
It seems like this Return Completion could be returned by `IteratorClose`, and thus by this algorithm, and thus by the `[[Call]]` for this built-in. But the invariants for essential internal methods don't allow `[[Call]]` to return a Return Completion.